### DISCUSSION TITLE
Does anyone use next.js SSR with Firestore?

### DISCUSSION THREAD

**Discussion by ivanhoe-dev** [vercel/next.js#25439] [Help]
Im currently developing a Next.js application with Firebase's Firestore as storage solution. As SEO is a concern for me, I would like to fetch data in run time and render it from the server side. However, I found there are rare examples for this implementation. Seems the client Firebase library not working in the server side. Im wondering if anyone has such experience on Next.js SSR and Firestore(AdminSDK)? Thanks!

**Comment by wwwdepot** [comment]
Checkout the [with-firebase](https://github.com/vercel/next.js/tree/canary/examples/with-firebase) example, its profile page is using `getServerSideProps` SSR

**Reply by ivanhoe-dev** [reply]
Thanks, i just checked the example, everything is working until it get the firestore record in SSR. The application hangs and not able to get any result. May I know is it the same in your side?

**Reply by wwwdepot** [reply]
The example is working on my side. Look for errors in the network and console tab.

**Reply by ivanhoe-dev** [reply]
Hi @wwwdepot I tried to debug, but the console shows nothing because the issue happens in server side. In the network tab, it shows the page is forever pending but no response. May I know is the /profile/[username] SSR working in your side? Thanks

Screenshot:
https://ibb.co/MsDWpNY
